mesh/pmesh.cpp

```diff
@@ -6678,6 +6678,122 @@ void ParMesh::GetGlobalElementIndices(Array<HYPRE_BigInt> &gi) const
    }
 }
 
+void ParMesh::GetExteriorFaceMarker(Array<int> & face_marker) const
+{
+   const_cast<ParMesh*>(this)->ExchangeFaceNbrData();
+
+   Mesh::GetExteriorFaceMarker(face_marker);
+}
+
+void ParMesh::UnmarkInternalBoundaries(Array<int> &bdr_marker, bool excl) const
+{
+   const int max_bdr_attr = bdr_attributes.Max();
+
+   MFEM_VERIFY(bdr_marker.Size() >= max_bdr_attr,
+               "bdr_marker must be at least bdr_attriburtes.Max() in length");
+
+   Array<int> ext_face_marker;
+   GetExteriorFaceMarker(ext_face_marker);
+
+   Array<bool> interior_bdr(max_bdr_attr); interior_bdr = false;
+   Array<bool> exterior_bdr(max_bdr_attr); exterior_bdr = false;
+
+   // Identify attributes which contain local interior faces and those which
+   // contain local exterior faces.
+   for (int be = 0; be < boundary.Size(); be++)
+   {
+      const int bea = boundary[be]->GetAttribute();
+
+      if (bdr_marker[bea-1] != 0)
+      {
+         const int f = be_to_face[be];
+
+         if (ext_face_marker[f] > 0)
+         {
+            exterior_bdr[bea-1] = true;
+         }
+         else
+         {
+            interior_bdr[bea-1] = true;
+         }
+      }
+   }
+
+   Array<bool> glb_interior_bdr(bdr_attributes.Max()); glb_interior_bdr = false;
+   Array<bool> glb_exterior_bdr(bdr_attributes.Max()); glb_exterior_bdr = false;
+
+   MPI_Allreduce(&interior_bdr[0], &glb_interior_bdr[0], bdr_attributes.Max(),
+                 MPI_C_BOOL, MPI_LOR, MyComm);
```

Comment:
I know it is an old story, but is `MPI_C_BOOL` the right type? 🤔  It should be used for C `_Bool` type. For C++ `bool`, there is `MPI_CXX_BOOL` from MPI 2.2. However, both [seem to be fixed](http://www.unixer.de/blog/index.php/2009/07/30/the-mpi-standard-mpi-22-is-fixed-now/) and have size one 😌 .

Reply:
Fair question. We use this same pattern four other times in `pmesh.cpp` so that's what I followed. However, `MPI_CXX_BOOL` does seem more correct.

Reply:
As discussed, this for a follow-up and would be better to do consistently everywhere. Leaving this not resolved for a future reference.